File: python/SCOOTER/Tests/DW/Arctic/arcticS.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('Scooter - Arctic profile') 

system('scooter.exe arcticS');
system('fields.exe arcticS < arcticS.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'arcticS.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

tej = flipud( jet( 256 ) );

[ ~, ~, ~, ~, Pos, G ] = read_shd( 'arcticS.grn' );

G = squeeze( G );
k = Pos.r.range; 
z = Pos.r.depth;

figure(1)
pcolor( k,-z,abs(G) ), shading interp, colormap( jet ), caxis([0 50]), colorbar
xlabel('Wavenumber (1/m)')
ylabel('Depth (m)')
title('SCOOTER - Green function, 200 Hz')

figure(2), hold on
plot(0,zs,'ko',0,zs,'m*','MarkerSize',16) 
pcolor(rarraykm,zarray,tl), shading interp, colormap( tej ), caxis([40 90])
h = colorbar;
set( h, 'YDir', 'reverse' );

axis([0 rmaxkm 0 zmax])
hold off
view(0,-90)
xlabel('Range (km)')
ylabel('Depth (m)')
title('SCOOTER - Arctic profile, 200 Hz')

disp('done.')

File: python/SCOOTER/Tests/DW/Arctic/arcticS.py
#==================================================================
#  
#  KRAKEN: Arctic sound speed profile
#  Mexilhoeira Grande, Sex Jul 28 18:37:28 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run arcticS

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("SCOOTER- Arctic profile")

system("scooter.exe arcticS")
system("fields.exe arcticS < arcticS.flp")

filename = 'arcticS.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure, axis=(0,1) )
tl = -20*log10( abs( p ) )

figure(1)
imshow(tl,extent=[0,rmaxkm,0,Dmax],aspect='auto',cmap='jet_r',origin='lower',vmin=40,vmax=90)
cb = colorbar()
cb.ax.invert_yaxis()
plot(rs,-zs,marker="<",markersize=16,color="k")
xlabel('Range (km)')
ylabel('Depth (m)')
title('SCOOTER - Arctic waveguide')
ylim(Dmax,0)

show()

print("done.")
